packages/twenty-server/src/engine/core-modules/billing/services/billing-portal.workspace-service.ts

```diff
@@ -36,7 +36,9 @@ export class BillingPortalWorkspaceService {
     plan?: BillingPlanKey,
     requirePaymentMethod?: boolean,
   ): Promise<string> {
-    const frontBaseUrl = this.domainManagerService.getBaseUrl();
+    const frontBaseUrl = this.domainManagerService.getBaseUrl(
+      workspace.subdomain,
+    );
```

Comment:
```suggestion
    const frontBaseUrl = this.domainManagerService.buildWorkspaceURL({
      subdomain: workspace.subdomain
    });
```
You should use buildWorkspaceURL which is designed to manage this kind of use case.